snakemake/wrangler_by_sample_finish.smk

```diff
@@ -1,81 +1,159 @@
-'''
-creates a mip_ids folder and an allMipsSamplesNames.tab.txt file. extracts mips,
-corrects mips, and generates files that can be used to determine sample names as
-well as sample names that had extractable data.
-'''
-
 configfile: 'wrangler_by_sample.yaml'
-output=config['output_folder']
+output='/opt/analysis'
+
+all_samples, all_targets=[],[]
+
+for line_number, line in enumerate(open(output+'/mip_ids/allMipsSamplesNames.tab.txt')):
+	if line_number>0:
+		line=line.rstrip().split('\t')
+		if len(line)>1 and len(line[1])>0:
+			all_samples.append(line[1])
+		if len(line[0])>0:
+			all_targets.append(line[0])
+
+final_dict={1: expand(output+'/analysis/{sample}/{sample}_mipExtraction/log.txt', sample=all_samples),
+			2: expand(output+'/analysis/{sample}/{sample}_mipBarcodeCorrection/barcodeFilterStats.tab.txt', sample=all_samples),
+			3: output+'/analysis/logs/mipCorrectForContamWithSameBarcodes_run1.json',
+			4: expand(output+'/clustering_status/{sample}_mip_clustering_finished.txt', sample=all_samples),
+			5: expand(output+'/analysis/populationClustering/{target}/analysis/log.txt', target=all_targets),
+			6: output+'/allInfo.tsv.gz'}
+output_choice=config['output_choice']
+final_out=final_dict[output_choice]
 
 rule all:
 	input:
-		setup_finished=output+'/setup_finished.txt',
-#		good_samples=output+'/successfully_extracted_samples.txt',
-		output_configfile=output+'/snakemake_params/wrangler_by_sample.yaml'
+		final_out
 
-rule copy_files:
```

Comment:
I'm not following why the rule "copy_files" is not needed anymore. I think you (Charlie) explained this to me but I can't remember it. Is it because the setup run snakemake script already takes care of the copying?